src/common/collection_column_specs/parse_sample_template.py

```diff
@@ -0,0 +1,151 @@
+
+
+_NGRAM_KEY = ['other_names',
+              'purpose',
+              'sesar:collection_method_description',
+              'sesar:archive_contact_current',
+              'sesar:collector_chief_scientist']
+
+_DATE_KEY = ['modification_date']  # example from sample service cannot be parsed by datetime.strptime
+# description from sample service needs to be corrected
+_CUSTOM_DESCRIPTION = {'sesar:igsn': 'International Geo Sample Number',
+                       'longitude': 'Longitude of the location where the sample was collected in WGS 84 coordinate '
+                                    'system.',
+                       'latitude': 'Latitude of the location where the sample was collected in WGS 84 coordinate '
+                                   'system.',
+                       }
+# shared sample attributes with distinct display names across sample services (sesar and enigma).
+_CUSTOM_DISPLAY_NAME = {'longitude': 'Longitude',
+                        'latitude': 'Latitude',
+                        }
+
+
+def _is_date_string(example_value, key):
+    if key in _DATE_KEY:
+        return True
+
+    # https://github.com/kbase/collections/blob/main/src/loaders/common/loader_helper.py#L52
+    formats_to_try = ["%Y/%m/%d", "%Y-%m-%d", "%m/%d/%y", "%Y-%m-%dT%H:%M:%S%z"]
+    for date_format in formats_to_try:
+        try:
+            datetime.strptime(example_value, date_format)
+            return True
+        except ValueError:
+            pass
+
+
+def _string_type(example_value, key):
+    if isinstance(example_value, str):
+        # handle a situation like '2; 10'
+        # https://github.com/kbase/sample_service_validator_config/blob/master/templates/enigma_template.yml#L309C5-L309C19
+        example_value = example_value.split(';')[0].strip()
+        if _is_date_string(example_value, key):
+            return {"type": "date"}
+        try:
+            int(example_value)
+            return {"type": "int"}
+        except ValueError:
+            try:
+                float(example_value)
+                return {"type": "float"}
+            except ValueError:
+                return {"type": "string",
+                        "filter_strategy": "ngram" if key in _NGRAM_KEY else "identity"}
+    elif isinstance(example_value, int):
+        return {"type": "int"}
+    elif isinstance(example_value, float):
+        return {"type": "float"}
+    else:
+        raise ValueError(f'Unknown type for {example_value} for key {key}')
+
+
+def _parse_input_yaml(input_yaml):
+    with open(input_yaml, 'r') as f:
+        data = yaml.safe_load(f)
+
+    result = []
+
+    for column_name, column_data in data['Columns'].items():
+
+        if 'transformations' not in column_data:
+            key = column_data['aliases'][0]
+        else:
+            transformation_data = column_data['transformations'][0]
+            key = transformation_data['parameters'][0]
```

Comment:
I didn't make the sample's templates either. But by scanning through the file, I think this covers all of them samples defined.

Reply:
I just don't understand why we're getting the key from anything other than column_name

Reply:
hmmm. I didn't find any doc explaining the template format. But a typical template is something like this. We are looking for `enigma:date` in this case which is the `['transformations'][0]['parameters'][0]`.

```
  Date:
    aliases:
    - date
    category: collection
    definition: Date (YYYY-MM-DD)
    example: '2019-06-18'
    order: 8
    required: false
    transformations:
    - parameters:
      - enigma:date
      transform: map
```

Reply:
`column_name` appears to be the display name in the template.

Reply:
Hmm... is this a sample validation spec or something else? Or did the sample validation spec change since I worked on it?

Reply:
are you referring to this file?
https://github.com/kbase/sample_service_validator_config/blob/master/metadata_validation.yml

I didn't find this file before. Instead I used: https://github.com/kbase/sample_service_validator_config/blob/master/templates/enigma_template.yml

Reply:
ok, yeah, I'm familiar with the first file but not the 2nd

Reply:
Fair enough then

Reply:
:

Reply:
I added a comment in the script to use the `metadata_validation.yml` as an alternative method. I now think that file might be better structured. But I guess we don't need to update the script now since it's not affecting what we are doing for sample filtering.